src/GraphQL.SystemTextJson/StringExtensions.cs

```diff
@@ -29,25 +31,20 @@ public static class StringExtensions
         /// <param name="json">A JSON formatted string.</param>
         /// <returns>Inputs.</returns>
         public static Inputs ToInputs(this string json)
-        {
-            var dictionary = json?.ToDictionary();
-            return dictionary.ToInputs();
-        }
+            => json != null ? JsonSerializer.Deserialize<Inputs>(json, _jsonOptions) : Inputs.Empty;
 
         /// <summary>
         /// Converts a JSON-formatted string into a dictionary of objects of their actual type.
         /// </summary>
         /// <param name="json">The json.</param>
         /// <returns>Dictionary.</returns>
+        [Obsolete("This class will be removed in a future version of GraphQL.NET. Please use the ToInputs method instead.")]
         public static Dictionary<string, object> ToDictionary(this string json)
             => JsonSerializer.Deserialize<Dictionary<string, object>>(json, _jsonOptions);
 
         /// <summary>
         /// Deserializes a JSON-formatted string of data into the specified type.
         /// <br/><br/>
-        /// Any <see cref="Dictionary{TKey, TValue}">Dictionary&lt;string, object&gt;</see> objects will be deserialized
-        /// into the proper format for passing to <see cref="InputsExtensions.ToInputs(Dictionary{string, object})"/>.
-        /// <br/><br/>
```

Comment:
This method was just added in 4.4.0 (released yesterday) - part of PR #2521.  I suggest removing this comment since we intend to remove this functionality in 5.0 (assuming we merge this PR).

Also, as a breaking change from 4.4.0 (released yesterday), we could disable the `ObjectDictionaryConverter` for this method now.... but I did not do so.  If we release 4.5.0 immediately, I would make the breaking change.  If not, then we can just note it in the 5.0 release notes.